### PR TITLE
Fixed more warnings about extra semicolons

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

**Review by mattjala — APPROVED** [review]

File: src/H5Tconv.c
Comment (jhendersonHDF) [inline src/H5Tconv.c:1]: Looks like we'll have to convert all the H5T_CONV_xx macros to do..while form at some point to make clang format happy again in this file

**Review by jhendersonHDF — APPROVED** [review]

**Review by qkoziol — APPROVED** [review]
Would be nice to get the formatting in H5Tconv.c straightened out...  :-/

**Comment by lrknox** [comment]
This PR didn't run any tests, and since there were so many files changed I ran configure and make with a few different options.  There were errors with MPICH 3.3 and configure --enable-fortran --enable-parallel.  I don't think fortran is an issue, but parallel.  

Summary - 64 "error:" messages for these 4 files:
        H5Centry.c
	H5Cimage.c
	H5Cpkg.h
	H5private.h

Some of the common errors:
../../seanm-hdf5/src/H5Cpkg.h:1682:5: error: expected ‘;’ before ‘if’
../../seanm-hdf5/src/H5Cpkg.h:1687:7: error: ‘else’ without a previous ‘if’
../../seanm-hdf5/src/H5Cpkg.h:1696:5: error: expected ‘;’ before ‘}’ token
../../seanm-hdf5/src/H5Centry.c:350:9: error: expected ‘while’ before ‘if’
../../seanm-hdf5/src/H5Cpkg.h:1422:1: error: expected ‘;’ before ‘}’ token
../../seanm-hdf5/src/H5Centry.c:708:5: error: expected ‘while’ before ‘else’
../../seanm-hdf5/src/H5Centry.c:922:1: error: invalid storage class for function ‘H5C__vea’ (and other functions)
../../seanm-hdf5/src/H5Centry.c:2339:21: error: label ‘done’ used but not defined
../../seanm-hdf5/src/H5Centry.c:4214:1: error: expected declaration or statement at end of input
 } /* H5C__remove_entry() */

I suspect the first few in H5Cpkg.h may have to do with the "#ifdef H5_HAVE_PARALLEL" ... #else ... #endif code from lines 1650 - 1667.

**Comment by seanm** [comment]
@lrknox when I try to turn on `HDF5_ENABLE_PARALLEL` in cmake, I get:

```
 Checking for module 'mpi-c'
   No package 'mpi-c' found
```

Is it talking about this https://www.open-mpi.org/ ?

**Comment by seanm** [comment]
To answer my own question, yes, it's that library. I can build with that option now, and have updated the patch.

**Review by lrknox — APPROVED** [review]
Various configurations build now and all tests passed.

**Review by brtnfld — APPROVED** [review]
Fortran looks good, thanks!

**Pull request merged**